Add <html> tag and language

jszwedko opened the pull request:
WCAG 2.0, Guideline 3.1.1 requires that

"The default human language of each Web page can be programmatically
determined"

Discovered this while doing an accessibility audit of our Jenkins site and
tripped the
https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_html_01
rule of Google's automated accessibility tool.

pathawks commented:
There is exactly one word on the page.

What actual benefit is there to setting the language?

benbalter commented:
Is there any downside? Not to be too pedantic, but to avoid the obvious follow up issue, perhaps it should be `{{ site.lang | default: "en-us" }}`?

jszwedko commented:
I thought about that @benbalter, but since this page is strictly in English (even if the rest of the author's site is not), it felt like `en-US` was the most appropriate. I'm curious if you have more thoughts on this though.

@pathawks You can see some of the benefits of setting the default language for users using assistive technologies on https://www.w3.org/TR/UNDERSTANDING-WCAG20/meaning-doc-lang-id.html. Typically the user would never see this page as it should immediately redirect, but the benefit is for users where the page doesn't do this automatically (if they have JS disabled, e.g.). I agree that there isn't much text content on the page, but I don't see any obvious downsides to setting the language even if there was only 1 word (in this case there is 9 including the page title). I may be missing something though, please let me know if so.

I saw the same test failures existing in Travis, locally, on master, but was hoping it was just something with my environment. I'll take a closer look.

jszwedko commented:
I think the test failures are related to https://github.com/jekyll/jekyll-redirect-from/issues/99

jszwedko commented:
And are fixed by https://github.com/jekyll/jekyll-redirect-from/pull/96

pathawks commented:
So the advantage would be that, if a user does not speak English *and* the browser cannot automatically redirect the user, some browser plugin might automatically translate the page into the users native language?

Makes sense.

jszwedko commented:
@pathawks the intent is to serve visitors with disabilities that rely on technology like screen readers to browse the web. I'm less familiar with automatic language translation, but I imagine that, yes, it may also assist with that.

pathawks commented:
@jszwedko I see where you're coming from and, while I don't disagree, I'm not sure what circumstances would cause the page to not redirect automatically.

jszwedko commented:
@pathawks the primary case in which this would happen is where the user does not have JS enabled (either by default or in general) -- I imagine this use case is also why there is link on the page for users to click. Given that there is text, I don't think it is unreasonable to set the language for the page :smile:

pathawks commented:
No, not unreasonable.

I *would* note that, even when JS is disabled, the `<meta>` tag should still trigger a redirect.

jszwedko commented:
That is fair, missed that!

parkr commented on lib/jekyll-redirect-from/redirect_page.rb:
```diff
@@ -21,6 +21,7 @@ def initialize(site, base, dir, name)
     def generate_redirect_content(item_url)
       self.output = self.content = <<-EOF
 <!DOCTYPE html>
+<html lang="en-US">
```
I'm ok if we add this. Could we terminate it on line 32?

jszwedko replied:
:+1:

jszwedko replied:
https://github.com/jekyll/jekyll-redirect-from/pull/100/commits/bddec45cfc2635b9127ecb100e32adfb9a56c3b7

parkr replied:
@jszwedko Did you consider using @benbalter's suggestion?

> `{{ site.lang | default: "en-US" }}`

jszwedko replied:
@parkr I did, but since the text on the page is English (regardless of what the site language is), it felt like `en-US` was the best fit. Curious to hear any other thoughts about this though.

parkr commented:
@jszwedko Your tests need to be updated to account for this change.

jszwedko commented:
@parkr doh, that's what I get for editing through the Github UI. I updated the test and squashed the commits down to one. Thanks for pointing that out!

pathawks commented:
:shipit:

parkr commented:
:shrug: ok!

benbalter commented:
@jekllybot: merge

benbalter commented:
Thanks @jszwedko! Sorry this took so long to merge :smile:

jszwedko commented:
@benbalter no worries, thanks for getting it in!
